### ISSUE TITLE
Difficulty Replicating Code

### ISSUE THREAD

**Issue by Evvvvvzen** [runjia0124/CoCoNet#4]
Hello @runjia0124 

I've been working with your code in CoCoNet and I've encountered some difficulties while trying to replicate the results. 
Specifically, 

Issue 1: Package Version Problems
I've noticed that there might be some problems with the package versions in the requirement.txt file, like "etelemetry==0+unknown" or leading to compatibility issues. I've tried manually changing these versions, but I'm still facing difficulties. Do you have any recommended package versions or solutions?

Issue 2: GPU Activation Problem
While I've ensured that my PyTorch version matches yours, when using the --test parameter, I've noticed that the GPU doesn't seem to be activated (cuda False), and the training mode is also not enabled (training False). I've verified the CUDA installation and GPU drivers, but I haven't been able to resolve the issue. Are there any specific settings or suggestions for enabling GPU usage in test mode?

I'm wondering if you could provide some insights or guidance on how to overcome this challenge. Any help would be greatly appreciated.

Thank you!

**Comment by runjia0124** [comment]
Hello @Evvvvvzen ,

Thanks for reporting this,

Issue 1: Apologies for the problem. Please try the updated [requirements.txt](https://github.com/runjia0124/CoCoNet/blob/main/requirements.txt). If you run into other conflicts, simply try removing its version string. 

Issue 2: The '--use_gpu' flag should activate CUDA usage. 'torch.cuda.is_available()' returning False means CUDA is not used by torch. Could you first make sure you have torch and CUDA configured correctly? (e.g., did you install pytorch-gpu or pytorch-cpu?) For the training mode, it is supposed to be False since you are in a testing mode.

Feel free to reach out if you have more questions,

Best,
Runjia

**Comment by Evvvvvzen** [comment]
Hello @runjia0124 
Thank you for your response, and I appreciate the solution provided for the updated "requirements.txt." I have confirmed that it is working well.

Regarding Issue 2, I have already verified that my CUDA and PyTorch installations are correct, and "torch.cuda.is_available()" returns True, indicating that CUDA can be used by PyTorch. 

However, even in this scenario, when I run the following command:
`python main.py --test --use_gpu --test_vis ./TNO/VIS --test_ir ./TNO/IR`

I still encounter issues and cannot activate the GPU correctly. Do I need to further configure additional environment variables or options to ensure the GPU is properly activated?

Once again, thank you for your assistance.

![image](https://github.com/runjia0124/CoCoNet/assets/96003076/d286b5b0-3e54-44bc-b46f-95cb22b17fa7)
![image](https://github.com/runjia0124/CoCoNet/assets/96003076/d7b2ad0c-4e25-43bb-ac10-727083ddaaf6)

**Comment by Evvvvvzen** [comment]
Hello @runjia0124,

I've identified the issue. It's in line 16 of the 'main.py' file where `'os.environ["CUDA_VISIBLE_DEVICES"] = '1'` is set. (turn 1 to 0 it's ok) This setting prevents CUDA from detecting the GPU because my server has only one GPU. 

Additionally, my GPU is an RTX 4080, so I need to install a higher version of Torch to make it work.

In closing, I want to express my appreciation for your contribution to the open-source community by sharing your code.

Best regards,
Evzen

**Comment by runjia0124** [comment]
Hello @Evvvvvzen ,

I appreciate your efforts! I will update the hardcoded CUDA index to avoid future issues. Thank you for reporting back!
